### DISCUSSION TITLE
Defining invoked machine actions outside of declaration.

### DISCUSSION THREAD

**Discussion by TeraBlitz** [statelyai/xstate#1296] [General] [answered]
Is there any way of defining the actions of an invoked machine outside of the declaration of said machine on react.
Example:
```
let stateM2 =Machine({
  initial: 'x11',
  states:{
    x11: {
      on: {
        goto:{
          target: 'x12',
        }
      }
    },
    x12: {
      onEntry: 'example',
      on: {
        goto:{
          target: 'x13'
        }
      }
    },
    x13: {
      type:'final',
      data: {
        performedAction: 'algo',  
      }
    }
  },
})
const stateMachine = Machine({
  id: 'form',
  initial: 'x0',
  context: {
    test: ''
  },
  states: {
    x0: {
      on: {
        goto:{
          target: 'x1',
        }
      }
    },
    x1: {
      on:{
        goto:{
          actions: send('goto', {to: 'child'})
        }
      },
      invoke: {
        id:'child',
        src: stateM2,
        onDone: {
          target:'x2',
        }
    }
    },
    x2: {
      on: {
        goto: {
          target: 'x0',
        }
      }
    }
  }
})
```
Id like to declare what the action `example` does outside of this declaration.
I tried declaring this action on its parent using `withConfig`:
```
const [state, send] = useMachine(stateMachine.withConfig({
    actions:{
      example: () => {console.log('exampel work')}
    }
  }))
```
but this does not work (as it shouldn't) but then is there a way of doing `withConfig` with its invoked machines?.

**Comment by davidkpiano** [comment, marked as answer]
You don't want to cross actor boundaries like that. The more reasonable thing to do would be to delegate the execution of that action to the **parent**, since that is where it's most relevant. And then the child actor can send events to the parent, and the parent can trigger actions (or not) based on that event.
